### PR TITLE
HAWQ-1420: Enable build of Ranger plugin open source RPM

### PR DESCRIPTION
**Pull request by ljainpivotalio**

### CODE REVIEW

**Comment by radarwave** [comment]
Does this fixed below issue? Apache hawq rpm name should be 'apache-hawq', not 'hawq_2_2_0_0'.

rpm -ivh hawq-ranger-plugin_2_2_0_0-2.2.0.0-1.el7.centos.noarch.rpm
error: Failed dependencies:
	hawq_2_2_0_0 >= 2.2.0.0 is needed by hawq-ranger-plugin_2_2_0_0-2.2.0.0-1.el7.centos.noarch

**Comment by ljainpivotalio** [comment]
No. The dependency on hawq needs to change to apache-hawq or must be excluded in branch for the open source binary build.

**Comment by radarwave** [comment]
@ljainpivotalio So we need another jira to fix the dependence issue?

File: ranger-plugin/pom.xml [hunk 1]
```diff
@@ -38,6 +38,9 @@
         <release.version>1</release.version>
         <postgresql.version>9.1-901-1.jdbc4</postgresql.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <build.suffix>_${hawq.name.version}</build.suffix>
+        <hawq.dep.name>hawq_${hawq.name.version}</hawq.dep.name>
```
Comment (radarwave) [inline ranger-plugin/pom.xml]: hawq.dep.name should be something like 'apache-hawq'.

File: ranger-plugin/pom.xml [hunk 2]
```diff
@@ -38,6 +38,9 @@
...
+        <destination.dir>/usr/local/hawq${build.suffix}/ranger</destination.dir>
```
Comment (radarwave) [inline ranger-plugin/pom.xml]: Apache HAWQ will default get installed into /usr/local/apache-hawq.

**Comment by ljainpivotalio** [comment]
@radarwave The open source build can override these values using maven command line:

BUILD_OPTS="-Drelease.version=bin -Dbuild.suffix= -Dhawq.dep.name=apache-hawq  -Ddestination.dir=/usr/local/apache-hawq"

**Comment by radarwave** [comment]
Now hawq ranger plugin rpm is in different directory, we should update the  tarball script 'contrib/hawq-package/make_rpm_tarball.sh' with the new copy directory:

-cp ${SRC_TOP_DIR}/ranger-plugin/target/rpm/hawq-ranger-plugin_*/RPMS/noarch/hawq-ranger-plugin*.rpm ${RPM_PKG_DIR}/
+cp ${SRC_TOP_DIR}/ranger-plugin/target/rpm/hawq-ranger-plugin*/RPMS/noarch/hawq-ranger-plugin*.rpm ${RPM_PKG_DIR}/

**Comment by radarwave** [comment]
Now the rpm name is: hawq-ranger-plugin-2.2.0.0-1.el7.centos.noarch.rpm

How about we remove the string 'centos' since there is already a 'el7' string there?

**Comment by ljainpivotalio** [comment]
```
mvn ${MVN_OPTS} -N ${BUILD_OPTS} install
[INFO] Creating spec file /root/ranger-plugin/target/rpm/hawq-ranger-plugin/SPECS/hawq-ranger-plugin.spec
[INFO] Building target platforms: noarch-redhat-linux
[INFO] Building for target noarch-redhat-linux
[INFO] Processing files: hawq-ranger-plugin-2.2.0.0-1.el6.noarch
[INFO] Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
[INFO] Checking for unpackaged file(s): /usr/lib/rpm/check-files /root/ranger-plugin/target/rpm/hawq-ranger-plugin/buildroot
[INFO] Wrote: /root/ranger-plugin/target/rpm/hawq-ranger-plugin/RPMS/noarch/hawq-ranger-plugin-2.2.0.0-1.el6.noarch.rpm
[INFO] Executing(%clean): /bin/sh -e /var/tmp/rpm-tmp.EplE5i
[WARNING] + umask 022
[WARNING] + cd /root/ranger-plugin/target/rpm/hawq-ranger-plugin/BUILD
[WARNING] + /bin/rm -rf /root/ranger-plugin/target/rpm/hawq-ranger-plugin/buildroot
[WARNING] + exit 0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 30.102 s
[INFO] Finished at: 2017-03-31T23:37:12+00:00
[INFO] Final Memory: 14M/95M
[INFO] ------------------------------------------------------------------------
```

**Comment by radarwave** [comment]
LGTM. +1

**Comment by huor** [comment]
+1 for building ranger rpm for apache hawq

**Pull request closed without merging**